src/gatsby-node.js

```diff
@@ -37,8 +37,9 @@ exports.sourceNodes = async (
   while (1) {
     itemsList.push(...data.items);
 
-    if (data.batching.next) {
-      data = await fetchData(data.batching.next);
+    if (data.batching) {
+      if (data.batching.next) data = await fetchData(data.batching.next);
+      else break;
     } else {
       break;
     }
```

Comment:
I've been searching around for a better of checking if a property of an object inside an object exists, and couldn't find anything neater so I went with the method I knew. Do recommend if you guys know better ways so I can refactor this. @datakurre @cekk

Reply:
There’s only more verbose ones. CoffeeScript had neat ?. test.